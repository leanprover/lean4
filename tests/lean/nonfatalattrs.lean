-- Even if there are errors in the attributes,
-- the definition should still be declared.

-- Attributes without implementation
syntax "unimplementedattr" : attr
@[unimplementedattr] def foo := 42
#check foo

-- Attributes that fail during application
def bar :=
  let rec @[class] bar : Nat := 42
  bar
#check bar
