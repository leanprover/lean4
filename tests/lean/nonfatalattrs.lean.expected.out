nonfatalattrs.lean:6:2-6:19: error: unknown attribute [attrUnimplementedattr]
foo : Nat
nonfatalattrs.lean:11:12-11:17: error: invalid 'class', declaration 'bar.bar' must be inductive datatype, structure, or constant
bar : Nat
